test/helpers/utils.go

```diff
@@ -247,6 +247,24 @@ func (kubectl *Kubectl) WaitCiliumEndpointReady(podFilter string, k8sNode string
 	return ciliumPod, endpoints
 }
 
+// WaitKafkaBroker waits for the broker to be ready, by executing
+// a produce request on existing topics and waiting for a response from broker
+func WaitKafkaBroker(pod string, cmd string, node *SSHMeta) error {
```

Comment:
TBH, I would move this function into `RuntimeValidatedKafka` test, like: 

```
WaitKafkaBroker := func(container, cmd) error {
	body...
}
```

Is a common pattern that we have followed, logger can be used, and that function is only used in KafkaRuntimeTest. Kubernetes Kafka tests need a different helper function, so if you can move into the test it'll be great

Example: 
https://github.com/cilium/cilium/blob/master/test/runtime/monitor.go#L78-L82

Reply:
I was planning to ping you too. I don't think I have used `WithTimeout` correctly, because the test seems to be hanging.

Reply:
done